tests/ignite/conftest.py

```diff
@@ -333,3 +336,159 @@ def _gloo_hvd_execute(func, args, np=1, do_init=False):
 @pytest.fixture()
 def gloo_hvd_executor():
     yield _gloo_hvd_execute
+
+
+skip_if_has_not_native_dist_support = pytest.mark.skipif(
+    not idist.has_native_dist_support, reason="Skip if no native dist support"
+)
+skip_if_has_not_xla_support = pytest.mark.skipif(not idist.has_xla_support, reason="Skip if no PyTorch XLA package")
+skip_if_has_not_horovod_support = pytest.mark.skipif(
+    not idist.has_hvd_support, reason="Skip if no Horovod dist support"
+)
+
+
+is_horovod_stash_key = pytest.StashKey[bool]()
+is_xla_stash_key = pytest.StashKey[bool]()
+is_xla_single_device_stash_key = pytest.StashKey[bool]()
+
+
+@pytest.fixture(
+    scope="module",
+    params=[
+        pytest.param(
+            "nccl_gpu",
+            marks=[
+                pytest.mark.distributed,
+                skip_if_has_not_native_dist_support,
+                pytest.mark.skipif(torch.cuda.device_count() < 1, reason="Skip if no GPU"),
+            ],
+        ),
+        pytest.param("gloo_cpu_or_gpu", marks=[pytest.mark.distributed, skip_if_has_not_native_dist_support]),
```

Comment:
Let's split `gloo_cpu_or_gpu` into `gloo_cpu` and `gloo` ?

Reply:
Do you mean `gloo_cpu` and `gloo_gpu`?

Reply:
No, gloo and gloo_cpu. By default nccl and gloo are considered for gpu